usagov/vote-gov-drupal
# VOTE-1350, VOTE-1351: Config split configuration

**_Delete all details that do not apply to this PR!_**

## Jira ticket (required)
[VOTE-1350](https://bixal-projects.atlassian.net/browse/VOTE-1350)
[VOTE-1351](https://bixal-projects.atlassian.net/browse/VOTE-1351)
## Description (optional)
Install and configure config_sync module.
New config dirs creates
* production
* non_prodcution
* local
* sync
The `sync` folder is now your default folder. The languages are now also compiled into the `sync` folder.

### Notable file changes
* .lando/settings.local.php - added in local config var
* docs/backend.md - created config_sync docs
* web/sites/default/settings.cloudgov.php - added in config_split vars
* web/sites/default/settings.php - set the config_split initial state
* composer.json - installed config_split
## Deployment and testing (required, if applicable)
### Pre-deploy steps
1. in `web/sites/settings.local.php` add the following to the bottom of the file
```
$settings['config_sync_directory'] = '../config/sync';
$config['config_split.config_split.local']['status'] = True;`
```

### Post-deploy steps
1. `drush cr`
2. `drush cim`

### Testing steps
1. After post-deployment, you can navigate to `/admin/config/development/configuration/config-split` and should see local as active.
![image](https://github.com/usagov/vote-gov-drupal/assets/17855589/7dac60da-c219-4cb2-a454-45ddc06a357e)
2. Navigate to `/admin/config/people/autologout` shoul dnotice extended logout time on local and on other env it the original timeout.
![image](https://github.com/usagov/vote-gov-drupal/assets/17855589/b7fbd234-ca33-4af2-b8e7-4f8dae736c6a)
3. to test other splits (nothing in them now... you could check it reverts settings.) chage this variable to proper machine name `$config['config_split.config_split.local']['status'] = True;`  ` drush cr` and then `drush cim`
